### PR TITLE
pre-commit: fix broken npm package files

### PR DESCRIPTION
**Pull request by avanthakkar**
Signed-off-by: Avan Thakkar <athakkar@redhat.com>

### CODE REVIEW

**Comment by nizamial09** [comment]
I remember the bot was complaining about this in one of my previous PR. Could you explain how its broken? @avanthakkar

**Comment by avanthakkar** [comment]
> I remember the bot was complaining about this in one of my previous PR. Could you explain how its broken? @avanthakkar

+ NPM_PACKAGE_FILES=0

> I remember the bot was complaining about this in one of my previous PR. Could you explain how its broken? @avanthakkar

Well with `grep -c` the script exits after `NPM_PACKAGE_FILES=0`, not sure why. So I think we can ignore bot suggestion here.
```
echo 'Running pre-commit hook...'
+ source /docker/ci/sanity-checks.sh
++ set -e
++ REPO_DIR=/ceph
++ [[ 1 == 1 ]]
++ [[ 17 -le 14 ]]
++ PYTHON_VERSION=3
++ TRANSLATION_FILE=src/pybind/mgr/dashboard/frontend/src/locale/messages.xlf
++ return
++ git diff --cached --name-only --diff-filter=ACMRTUXB
++ grep -Ec 'package(-lock){0,1}.json'
+ NPM_PACKAGE_FILES=0
```

**Comment by epuertat** [comment]
> > I remember the bot was complaining about this in one of my previous PR. Could you explain how its broken? @avanthakkar
> 
> * NPM_PACKAGE_FILES=0
> 
> > I remember the bot was complaining about this in one of my previous PR. Could you explain how its broken? @avanthakkar
> 
> Well with `grep -c` the script exits after `NPM_PACKAGE_FILES=0`, not sure why. So I think we can ignore bot suggestion here.
> 
> ```
> echo 'Running pre-commit hook...'
> + source /docker/ci/sanity-checks.sh
> ++ set -e
> ++ REPO_DIR=/ceph
> ++ [[ 1 == 1 ]]
> ++ [[ 17 -le 14 ]]
> ++ PYTHON_VERSION=3
> ++ TRANSLATION_FILE=src/pybind/mgr/dashboard/frontend/src/locale/messages.xlf
> ++ return
> ++ git diff --cached --name-only --diff-filter=ACMRTUXB
> ++ grep -Ec 'package(-lock){0,1}.json'
> + NPM_PACKAGE_FILES=0
> ```

That's easy @avanthakkar. The `set -e` will basically exit the script if any the last command run in a line returns non-zero (success). Not surprisingly `grep -c` returns non-zero when it doesn't match anything. The way to fix that is `... | grep -c || true`, which will always return 0.

**Comment by avanthakkar** [comment]
> > > I remember the bot was complaining about this in one of my previous PR. Could you explain how its broken? @avanthakkar
> > 
> > 
> > 
> > * NPM_PACKAGE_FILES=0
> > 
> > > I remember the bot was complaining about this in one of my previous PR. Could you explain how its broken? @avanthakkar
> > 
> > 
> > Well with `grep -c` the script exits after `NPM_PACKAGE_FILES=0`, not sure why. So I think we can ignore bot suggestion here.
> > ```
> > echo 'Running pre-commit hook...'
> > + source /docker/ci/sanity-checks.sh
> > ++ set -e
> > ++ REPO_DIR=/ceph
> > ++ [[ 1 == 1 ]]
> > ++ [[ 17 -le 14 ]]
> > ++ PYTHON_VERSION=3
> > ++ TRANSLATION_FILE=src/pybind/mgr/dashboard/frontend/src/locale/messages.xlf
> > ++ return
> > ++ git diff --cached --name-only --diff-filter=ACMRTUXB
> > ++ grep -Ec 'package(-lock){0,1}.json'
> > + NPM_PACKAGE_FILES=0
> > ```
> 
> That's easy @avanthakkar. The `set -e` will basically exit the script if any the last command run in a line returns non-zero (success). Not surprisingly `grep -c` returns non-zero when it doesn't match anything. The way to fix that is `... | grep -c || true`, which will always return 0.

Ah yes, thanks!

**Review by nizamial09 — APPROVED** [review]

File: docker/ceph/ci/pre-commit.sh [hunk 1]
```diff
@@ -1,13 +1,13 @@
...
-set -e
+set -ex
 
 echo 'Running pre-commit hook...'
 
 # shellcheck disable=SC1091
 source /docker/ci/sanity-checks.sh
 
-NPM_PACKAGE_FILES=$(git diff --cached --name-only --diff-filter=ACMRTUXB | grep -E "package(-lock){0,1}.json" -c)
+NPM_PACKAGE_FILES=$(git diff --cached --name-only --diff-filter=ACMRTUXB | grep -E "package(-lock){0,1}.json -c" || true)
```
Comment (nizamial09) [inline docker/ceph/ci/pre-commit.sh]: the `-c` should be outside double-quotes right?
```suggestion
NPM_PACKAGE_FILES=$(git diff --cached --name-only --diff-filter=ACMRTUXB | grep -E "package(-lock){0,1}.json" -c || true)
```

Comment (epuertat) [reply docker/ceph/ci/pre-commit.sh]: Yup, it's an option of grep
```suggestion
NPM_PACKAGE_FILES=$(git diff --cached --name-only --diff-filter=ACMRTUXB | grep -Ec "package(-lock)?.json" || true)
```

**Pull request merged**